### PR TITLE
[Php55] Handle crash on curly parentheses delimiter on PregReplaceEModifierRector

### LINKED ISSUE #7635: System error: "Syntax error, unexpected ':', expecting ',' or ']' or ')':1"
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.14.8|

## Minimal PHP Code Causing Issue

https://getrector.org/demo/84d4b4d6-0565-44fc-8354-639bfc5631a7

## Expected Behaviour

I honestly have no idea.

### PR DESCRIPTION
**Pull request by samsonasik**
Given the following code:

```php
function run()
{
    $eval_links = '$link = clean_link("$1","$2")';

    echo preg_replace('{\[link:([^\|].*?)\|([^\]].*?)\]}ei',"$eval_links",$str);
}
```

it cause crash:

```bash
There was 1 error:

1) Rector\Tests\Php55\Rector\FuncCall\PregReplaceEModifierRector\PregReplaceEModifierRectorTest::test with data set #2 ('/Users/samsonasik/www/rector-...hp.inc')
PhpParser\Error: Syntax error, unexpected '}' on line 1

/Users/samsonasik/www/rector-src/vendor/nikic/php-parser/lib/PhpParser/ParserAbstract.php:318
/Users/samsonasik/www/rector-src/vendor/nikic/php-parser/lib/PhpParser/ParserAbstract.php:159
/Users/samsonasik/www/rector-src/vendor/nikic/php-parser/lib/PhpParser/Parser/Multiple.php:51
/Users/samsonasik/www/rector-src/vendor/nikic/php-parser/lib/PhpParser/Parser/Multiple.php:32
/Users/samsonasik/www/rector-src/src/PhpParser/Parser/SimplePhpParser.php:38
/Users/samsonasik/www/rector-src/rules/Php72/NodeFactory/AnonymousFunctionFactory.php:157
/Users/samsonasik/www/rector-src/rules/Php55/Rector/FuncCall/PregReplaceEModifierRector.php:105
```

Ref https://getrector.org/demo/d1c86304-9f2a-4614-b636-f9bb1ebdaa16

This PR try to resolve it 

Fixes https://github.com/rectorphp/rector/issues/7635

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @themrwilliams

**Comment by samsonasik** [comment]
It seems not really working, ref https://3v4l.org/P4mPQ#v5.2.17 vs https://3v4l.org/22DeW#v5.2.17, probably skip is the way to go.

**Comment by samsonasik** [comment]
I skipped https://github.com/rectorphp/rector-src/pull/3144/commits/8b0e53f3c646e0fc25eed954b852eb5e558bf4fc

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
Ok, I got the solution by verify `Encapsed` parts value which start with `{` and end with `}` https://github.com/rectorphp/rector-src/pull/3144/commits/b5756bfe535c3f2df811a9f2134191eb6a2fed28

**Comment by samsonasik** [comment]
Proven works 🎉 https://3v4l.org/fqdPZ#v5.4.45 vs https://3v4l.org/Zm16C#v5.5.38

File: rules-tests/Php55/Rector/FuncCall/PregReplaceEModifierRector/Fixture/skip_curly_parentheses_delimiter.php.inc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+function clean_link($a)
+{
+    return "change";
+}
+
+function run($str)
+{
+    $eval_links = '$link = clean_link("$0")';
+
+    return preg_replace_callback('{tst}i',function ($matches) use ($eval_links) {
```
Comment (samsonasik) [inline rules-tests/Php55/Rector/FuncCall/PregReplaceEModifierRector/Fixture/skip_curly_parentheses_delimiter.php.inc]: oh, `e` inside regex should not be removed

Comment (samsonasik) [reply rules-tests/Php55/Rector/FuncCall/PregReplaceEModifierRector/Fixture/skip_curly_parentheses_delimiter.php.inc]:
```diff
-    return preg_replace_callback('{test}i',function ($matches) use ($eval_links) {
+    return preg_replace_callback('{tst}i',function ($matches) use ($eval_links) {
         return $link = clean_link($matches[0]);
     },$str);
```
The `e` part of is a regex value itself, ref https://github.com/rectorphp/rector-src/pull/3144/commits/e66e49f2ef37327d7672ede6bd7516ba3e76e539

Comment (samsonasik) [reply rules-tests/Php55/Rector/FuncCall/PregReplaceEModifierRector/Fixture/skip_curly_parentheses_delimiter.php.inc]: Fixed 🎉 https://github.com/rectorphp/rector-src/pull/3144/commits/652ceed5864929e87aeb5ff3ca1b1950c5c47c01

**Comment by samsonasik** [comment]
Finally, All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by samsonasik** [comment]
I added failing fixture e inside regex not as modifier https://github.com/rectorphp/rector-src/pull/3144/commits/27a6d4c9894e6e7afb7539b8711ae9aab8c0a6b6

**Comment by samsonasik** [comment]
Fixed 🎉 https://github.com/rectorphp/rector-src/pull/3144/commits/d150be8ae44057596d889e22ae5b551561914a51

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you 🙏

**Pull request merged**